R/replace_na.R

```diff
@@ -1,19 +1,31 @@
 #' Replace missing values
 #'
-#' @param data A data frame.
-#' @param replace A named list given the value to replace `NA` with
-#'   for each column.
+#' @param data A data frame or vector.
+#' @param replace The value(s) to replace `NA` with. If `data`
+#'   is a vector, a vector with one or more replacement values.
+#'   If `data` is a data.frame, a named list providing such
+#'   replacement vectors for each of the named columns.
 #' @param ... Additional arguments for methods. Currently unused.
 #' @examples
 #' library(dplyr)
 #' df <- tibble(x = c(1, 2, NA), y = c("a", NA, "b"))
 #' df %>% replace_na(list(x = 0, y = "unknown"))
+#' df$x %>% replace_na(3)
 #' @export
-replace_na <- function(data, replace = list(), ...) {
+replace_na <- function(data, replace = NULL, ...) {
   UseMethod("replace_na")
 }
 #' @export
-replace_na.data.frame <- function(data, replace = list(), ...) {
+replace_na.default <- function(data, replace = NULL, ...) {
+  if (length(replace) != 1L)
+    stop("If 'data' is a vector, 'replace' has to be of length 1, not ", length(replace))
+
+  data[are_na(data)] <- replace
```

Comment:
maybe I should use `replace[[1L]]` here to allow people to call `replace_na(vec, list(5))`?

Reply:
That's a good idea! It's more consistent with the rest of the API. (But you'll need to check that `replace()` is a length 1 list)

Reply:
We require it to be length 1 anyway, so no matter if list or not, `replace[[1L]]` will always be correct, no?

Reply:
I think it's better to be strict about the types here

Reply:
OK, done!